### ISSUE TITLE
manually have to run service.sh

### ISSUE THREAD

**Issue by notstevek** [Magisk-Modules-Repo/Custom-Notification-Snoozing#2]
As title says, if I want this to work, I have to manually run service.sh, otherwise snooze times are default, even though CNSValues.txt are changed to custom values. Only have to do this on system reboot. It holds up until a reboot.

Nexus 6p
Magisk 16.7
LineageOS 15.1 (OPM6.171019.030.H1)
ElementalX 6.08

**Comment by davidevertuani** [comment]
Same here.

Nexus 6P
Magisk 17.0
Stock Android 8.1.0
FrancoKernel r74

**Comment by HeyItsJono** [comment]
I've also started getting this issue recently, it might be that the script isn't snoozing for long enough anymore at boot-up before running, but what's weird is that the logs file is always empty, when it should show errors if that was the case. If you guys wouldn't mind, could you post your `/data/CNS.log` files for me? Also, when it does work, do you find that it resets to default values a few hours after booting?

I have a couple of solutions in mind for this - either extending the time it takes before running the `settings put global` command and hoping that fixes things, or maybe even including a build of crontab and having that run the script periodically. The latter would increase this module's overhead slightly and turn it from a script that only runs once at boot, to a script that runs every few hours or so, but it would prevent the issue of reversion back to default settings a while after boot. Since you guys are my target audience, what would you prefer? Do you think that's an okay trade-off, or would you rather take your chances with having it just run once at boot?

**Comment by notstevek** [comment]
I already have the crontab module installed, but to be honest, personally I don't even mind having to run it manually because I rarely reboot my phone, I was just reporting the issue, well, incase others were running into it and just so that you knew. My log file is empty. 0kb file. So personally, I would prefer that you don't include crontab since I use that module myself, having two is just, not something I'd prefer and just rather take the risk of running it once at boot, and if it works, great, if not, I run it manually. However, this is your module, and I can't tell you how to make it. If others want to have crontab included to fix it, go ahead, I'll just keep an older version. Thanks for taking the time out to speak directly to us and offer solutions.

**Comment by HeyItsJono** [comment]
No problem, I'll look into just getting it running properly at boot then, and if there's already a crontab module then for the time being, people could just use that to run it if necessary. Sorry it took so long to get back to you, I'm not on Github that often!

**Comment by notstevek** [comment]
I fixed it I think, I rebooted 6 times in a row with no issues, working everytime on my end. I'll put up a pull request and test it, see if it works for you too.

**Comment by HeyItsJono** [comment]
PR merged, and update pushed so it should now appear in the Magisk Manager app. Thanks again!

**Comment by notstevek** [comment]
Hey real quick, I read your readme, I'm not sure and I'm going to bed, but if we get rid of the "sleep 15" part or lower it, could get rid of the error of having to wait, or unless you have it set to sleep 15 for a specific reason. Anyway, as long as they don't go to set a snooze time until after the script kicks in, even notifications on boot should get the custom values. I know that I have a launcher notification immediately on boot, and it gets custom snooze times as long as I wait.

**Comment by HeyItsJono** [comment]
When I initially made the script, I found that `settings put global` was throwing errors if I didn't let the script sleep, and `sleep 15` was the lowest value I tested that worked at the time. This might have changed since, and in reality all of my notifications at boot now have custom times, but I put that warning in as a disclaimer. Might do some testing in the future to see if I can lower the `sleep` time without it breaking the script though!
